src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -266,13 +267,15 @@ private bool ShutdownHostIfUnhealthy()
         {
             if (ShouldMonitorHostHealth && _healthCheckWindow.GetEvents().Where(isHealthy => !isHealthy).Count() > _config.HostHealthMonitor.HealthCheckThreshold)
             {
-                // if the number of times the host has been unhealthy in
+                // If the number of times the host has been unhealthy in
                 // the current time window exceeds the threshold, recover by
-                // initiating shutdown
-                var message = $"Host unhealthy count exceeds the threshold of {_config.HostHealthMonitor.HealthCheckThreshold} for time window {_config.HostHealthMonitor.HealthCheckWindow}. Initiating shutdown.";
-                Instance?.TraceWriter?.Error(message);
+                // initiating a hard shutdown.
+                var message = $"Host unhealthy count exceeds the threshold of {_config.HostHealthMonitor.HealthCheckThreshold} for time window {_config.HostHealthMonitor.HealthCheckWindow}. Initiating hard shutdown.";
+                var shutdownEvent = new TraceEvent(TraceLevel.Error, message);
+                shutdownEvent.Properties.Add(ScriptConstants.TracePropertyEventNameKey, ScriptConstants.ShutdownRecoveryEventName);
```

Comment:
Adding a specific event that we can query on, to make it easier to review recovery situations in our system logs.